lib/src/main/java/graphql/nadel/hooks/NadelExecutionHooks.kt

```diff
@@ -106,7 +113,23 @@ interface NadelExecutionHooks {
     }
 
     fun partitionTransformerHook(): NadelPartitionTransformHook {
```

Comment:
Could also make this null, and omit the transform entirely if it's null

e.g.

```diff
diff --git a/lib/src/main/java/graphql/nadel/engine/plan/NadelExecutionPlanFactory.kt b/lib/src/main/java/graphql/nadel/engine/plan/NadelExecutionPlanFactory.kt
index 5608c3e0..ee6845f3 100644
--- a/lib/src/main/java/graphql/nadel/engine/plan/NadelExecutionPlanFactory.kt
+++ b/lib/src/main/java/graphql/nadel/engine/plan/NadelExecutionPlanFactory.kt
@@ -109,7 +109,9 @@ internal class NadelExecutionPlanFactory(
                 transforms = listOfTransforms(
                     NadelSkipIncludeTransform(),
                     NadelServiceTypeFilterTransform(),
-                    NadelPartitionTransform(engine, executionHooks.partitionTransformerHook()),
+                    executionHooks.partitionTransformerHook()?.let { partitionHook ->
+                        NadelPartitionTransform(engine, partitionHook)
+                    },
                     *transforms.toTypedArray(),
                     NadelDeepRenameTransform(),
                     NadelTypeRenameResultTransform(),
@@ -121,10 +123,10 @@ internal class NadelExecutionPlanFactory(
             )
         }
 
-        private fun listOfTransforms(vararg elements: NadelTransform<out Any>): List<NadelTransform<Any>> {
-            return elements.map {
+        private fun listOfTransforms(vararg elements: NadelTransform<out Any>?): List<NadelTransform<Any>> {
+            return elements.mapNotNull {
                 @Suppress("UNCHECKED_CAST") // Ssh it's okay
-                it as NadelTransform<Any>
+                it as NadelTransform<Any>?
             }
         }
     }
```

Reply:
This would make it harder to feature flag, as this code path is executed only during Nadel instance creation